daemon/ZilliqaDaemon.cpp

```diff
@@ -15,95 +15,152 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 
-#include <dirent.h>
-#include <errno.h>
-#include <signal.h>
-#include <stdlib.h>
-#include <sys/stat.h>
-#include <sys/types.h>
-#include <sys/wait.h>
-#include <unistd.h>
-#include <algorithm>
-#include <array>
-#include <cstdio>
-#include <fstream>
-#include <iostream>
-#include <memory>
-#include <stdexcept>
-#include <string>
-#include <unordered_map>
-#include <vector>
+#include "ZilliqaDaemon.h"
 
 using namespace std;
+namespace po = boost::program_options;
 
-const vector<string> programName = {"zilliqa", "lzilliqa", "latezilliqa"};
+#define SUCCESS 0
+#define ERROR_IN_COMMAND_LINE -1
 
-const string restart_zilliqa = "python tests/Zilliqa/daemon_restart_local.py";
+const vector<string> programName = {"zilliqa"};
+const string SYNCTYPE_OPT = "--synctype";
+const char* synctype_descr =
+    "0(default) for no, 1 for new, 2 for normal, 3 for ds, 4 for lookup, 5 "
+    "for node recovery, 6 for new lookup , 7 for ds guard node sync and 8 "
+    "for offline validation of DB";
 
-unordered_map<int, string> PrivKey;
-unordered_map<int, string> PubKey;
-unordered_map<int, string> Port;
-unordered_map<int, string> Path;
-unordered_map<int, bool> isDS;
+const string SUSPEND_LAUNCH = "SUSPEND_LAUNCH";
+const string upload_incr_DB_script = "upload_incr_DB.py";
+const string download_incr_DB_script = "download_incr_DB.py";
+const string auto_backup_script = "auto_backup.py";
+const string daemon_log = "daemon-log.txt";
 
 enum SyncType : unsigned int {
   NO_SYNC = 0,
   NEW_SYNC,
   NORMAL_SYNC,
   DS_SYNC,
   LOOKUP_SYNC,
+  RECOVERY_ALL_SYNC,
+  NEW_LOOKUP_SYNC,
+  GUARD_DS_SYNC,
+  DB_VERIF
 };
 
-const string logName = "epochinfo-00001-log.txt";
-
-string ReadLastLine(const string& filePath, [[gnu::unused]] ofstream& log) {
-  ifstream logFile;
+ZilliqaDaemon::ZilliqaDaemon(int argc, const char* argv[], std::ofstream& log)
+    : m_log(log),
+      m_logPath(boost::filesystem::current_path().string() + "/"),
+      m_curPath(boost::filesystem::current_path().string() + "/"),
+      m_port(-1),
+      m_recovery(0),
+      m_nodeIndex(0),
+      m_syncType(0),
+      m_cseed(false) {
+  if (ReadInputs(argc, argv) != SUCCESS) {
+    ZilliqaDaemon::LOG(m_log, "Failed to read inputs.");
+    exit(EXIT_FAILURE);
+  }
 
-  logFile.open(filePath + "/" + logName);
+  string msg = argv[0];
 
-  if (logFile.is_open()) {
-    logFile.seekg(-1, ios_base::end);
-  } else {
-    return "";
+  for (int i = 1; i < argc; ++i) {
+    msg += string(" ") + argv[i];
   }
 
-  bool keepLooping = true;
+  ZilliqaDaemon::LOG(m_log, msg);
+  KillProcess();
+  StartNewProcess();
+  StartScripts();
```

Comment:
I think we are missing something.  One's new process is spawned after being deleted, it should kill any running scripts and start scripts again. But its not the case right now.